kore/src/main/scala/org/kframework/compile/SubstitutionPruning.scala

```diff
@@ -0,0 +1,55 @@
+package org.kframework.compile
```

Comment:
I would like you to rewrite this code in Java. We are trying to reduce the amount of scala code we have in the code base, not increase it.

Reply:
Got it!  It wasn't clear to me so far (since my work was mostly on scala code lately) ...